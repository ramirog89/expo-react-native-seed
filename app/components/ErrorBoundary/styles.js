import { StyleSheet } from "react-native";
import { COLORS } from "../../constants/application";

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginVertical: 50
  },
  errorReport: {
    backgroundColor: COLORS.WHITE
  },
  infoReport: {
    backgroundColor: COLORS.WHITE
  }
});
