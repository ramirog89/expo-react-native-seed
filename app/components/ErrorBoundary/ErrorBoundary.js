import React from "react";
import { View, Text } from "react-native";
import PropTypes from "prop-types";

import styles from "./styles";

export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
      error: false,
      info: false
    };
  }

  static getDerivedStateFromError(error) {
    return { error, hasError: true };
  }

  componentDidCatch(error, info) {
    this.setState({
      hasError: true,
      error,
      info
    });
  }

  render() {
    if (this.state.hasError) {
      return (
        <View style={styles.container}>
          <View style={styles.errorReport}>
            <Text>Error: {this.state.error.name}</Text>
          </View>
          <View style={styles.infoReport}>
            <Text>Please try again</Text>
          </View>
        </View>
      );
    }

    return this.props.children;
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.any.isRequired
};
