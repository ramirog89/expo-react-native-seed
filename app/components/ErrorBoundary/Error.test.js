/** eslint-disable */
import React from "react";
import { View, Text } from "react-native";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import ErrorBoundary from "./ErrorBoundary";

class HandGranade extends React.Component {
  // eslint-disable-next-line react/require-render-return
  render() {
    throw (<Text>BOOOM!</Text>); // eslint-disable-line no-throw-literal
  }
}

describe("<ErrorBoundary />", () => {
  let Component;

  it("should render childs normally", () => {
    const rendered = renderer
      .create(
        <ErrorBoundary>
          <View>
            <Text>Hi</Text>
          </View>
        </ErrorBoundary>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it("should stop propagation of an error on render", () => {
    Component = mount(
      <ErrorBoundary>
        <HandGranade />
      </ErrorBoundary>
    );
    expect(Component.text()).toEqual("Error: Please try again");
  });
});
