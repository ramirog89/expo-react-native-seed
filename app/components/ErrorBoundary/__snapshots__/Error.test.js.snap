// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> should render childs normally 1`] = `
<View>
  <Text>
    Hi
  </Text>
</View>
`;
